lib/metadata/linux/LinuxSystemd.rb

```diff
@@ -0,0 +1,126 @@
+            @services.each do |service|
+              node = doc.add_element("service", service_xml(service))
+              service_targets_xml(service).each { |tx| node.add_element("target", tx) }
+            end
+        end
+
+        private
+
+        def parse_systemd
+            @services = files('.service').collect { |sf| parse_service(sf) }.compact
+            @targets  = files('.target').collect  { |tf| parse_target(tf)  }.compact
+        end
+
+        def files(unit_extension)
+          (SYSTEM_DIRS + USER_DIRS).flat_map do |dir|
+            dir_files = []
+            @fs.dirForeach(dir) do |unit|
+              dir_files << File.join(dir, unit) if @fs.fileExtname(unit) == unit_extension
+            end
+            dir_files
+          end
+        end
+
+        def ini(file)
+          fdata = nil
+          @fs.fileOpen(file) { |fo| fdata = fo.read }
+          IniParse.parse(fdata)
+        end
+
+        def parse_service(file)
+          return if @fs.fileSymLink?(file)
+          debug "Parsing service unit: #{file}"
+
+          unit        = @fs.fileBasename(file)
+          name        = unit.gsub(".service", "")
+          inif        = ini(file)
+          desc        = parse_description(inif)
+          wanted_by   = parse_wanted(inif)
+          required_by = parse_required(inif)
+
+          {:unit        => unit,
+           :name        => name,
+           :path        => file,
+           :description => desc,
+           :wanted_by   => wanted_by,
+           :required_by => required_by}
+
+          rescue
+            warn "Error parsing: #{file}"
+        end
+
+        def parse_description(inif)
+          return nil unless inif.has_section?('Unit') && inif['Unit'].has_option?('Description')
+          inif['Unit']['Description']
+        end
+
+        def parse_wanted(inif)
+          return [] unless inif.has_section?('Install') && inif['Install'].has_option?('WantedBy')
+          [inif['Install']['WantedBy']].flatten.collect(&:split).flatten
+        end
+
+        def parse_required(inif)
+          return [] unless inif.has_section?('Install') && inif['Install'].has_option?('RequiredBy')
+          [inif['Install']['RequiredBy']].flatten.collect(&:split).flatten
+        end
+
+        def parse_target(file)
+          return if @fs.fileSymLink?(file)
+          debug "Parsing target unit: #{file}"
+
+          unit = @fs.fileBasename(file)
+          name = unit.gsub(".target", "")
+          inif = ini(file)
+          desc = parse_description(inif)
+
+          {:unit        => unit,
+           :name        => name,
+           :path        => file,
+           :description => desc}
+          rescue
+            warn "Error parsing: #{file}"
+        end
+
+        def service_xml(service)
+          {"name"        => service[:name],
+           "path"        => service[:path],
+           "description" => service[:description],
+           "typename"    => "linux_systemd"}
```

Comment:
Before direct support for systemd is added to the UI, it would be nice if the current UI code could display the systemd entries as init processes - without the `'requiredby' / 'wantedby'` data, and leaving the `enable_run_levels` and `disable_run_levels empty`.

Given the change in `typename`, you probably have to change the association in the `VmOrTemplate` model to account for the new name. Currently, the association is defined as:
```
has_many  :linux_initprocesses, :class_name => "SystemService", :conditions => "typename = 'linux_initprocess'"
```
You'll probably need to change the `conditions` to something like"
```
..., :conditions => "typename = 'linux_initprocess' OR typename = 'linux_systemd'"
```
Going forward, we may want to define associations that distinguish between the two, as well as the union:
```
has_many :linux_legacy_initprocesses...
has_many :linux_systemd...
has_many :linux_initprocresses...
```
I guess the need for these will be determined by the new UI work.